lib/cocoapods/project.rb

```diff
@@ -25,6 +25,9 @@ def initialize(path, skip_initialization = false,
       @pods = new_group('Pods')
       @development_pods = new_group('Development Pods')
       self.symroot = LEGACY_BUILD_ROOT
+      root_object.build_configuration_list.build_configurations.each do |config|
```

Comment:
why do we care for the entire project to set this value? It seems that each and every other pod target that uses swift should use a value now.

Reply:
if you don't specify a value for the `SWIFT_VERSION` build settings, Xcode will not compile, so the idea here was to set a default version and allow each pod to override the default version as needed.

Reply:
@heyzooi doesnt each pod that uses swift automatically get assigned a version? I am still not 100% convinced we need to be changing the project configuration in this PR.